### DISCUSSION TITLE
Automating Hyperparameter Tuning in MONAI

### DISCUSSION THREAD

**Discussion by DevreeseJorik** [Project-MONAI/MONAI#6233] [Q&A]
Hi all,

I've been working with MONAI and have been trying to find information on how to automate hyperparameter tuning. I've searched through the documentation but couldn't find any mention of built-in tools or libraries that are compatible with MONAI for this purpose.

I'm wondering if anyone in the community has experience with automating hyperparameter tuning in MONAI or if there are any recommended libraries that can be used in conjunction with MONAI (and by extension, the SupervisedTrainer). This is not for any specific problem but a broad range of potential future projects, including but not limited to classification and segmentation. 

Any suggestions or insights on this topic would be greatly appreciated. Thank you!

**Comment by wyli** [comment]
in the auto3dseg solution it's possible to use nni/optuna https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/docs/hpo.md, please see also the auto3dseg tutorial in general https://github.com/Project-MONAI/tutorials/tree/main/auto3dseg

**Reply by DevreeseJorik** [reply]
Thank you for the prompt response and for pointing me towards the auto3dseg solution. I did come across the documentation mentioning the use of nni/optuna for hyperparameter tuning in this context.

However, I was wondering if this or a similar approach can be used with custom networks that have sizes other than 3D. Ideally I could run a bundle with any network and data shape.

Thank you again for your help and suggestions.
